### ISSUE TITLE
Deprecate tap2deb and tap2rpm

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7682]
|<img alt="jMyles's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jMyles reported|
|-|-|
|Trac ID|trac#7682|
|Type|defect|
|Created|2014-10-15 03:58:34Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-tap2whatever-7682|

tab2deb and tap2rpm are command-line utilities that create a shim over a .tap file, allowing it to be installed as an OS package (.deb or .rpm respectively).

Currently, a test (twisted.scripts.test.test_tap2deb.TestTap2DEB.test_basicOperation) fails in some situations, apparently because these utilities don't work properly in some specific OS configurations.  For example, see [#6920](https://github.com/twisted/twisted/issues/6920) (closed for other reasons).

It is not immediately obvious how widespread their use is, but exarkun advises deprecation.  Another developer will need to weigh-in with a deprecation justification is one is to be provided.

<details><summary>Searchable metadata</summary>

```
trac-id__7682 7682
type__defect defect
reporter__jMyles jMyles
priority__normal normal
milestone__ 
branch__branches_deprecate_tap2whatever_7682 branches/deprecate-tap2whatever-7682
branch_author__hawkowl__jMyles hawkowl, jMyles
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1413345514363505 1413345514363505
changetime__1429068576461375 1429068576461375
version__None None
owner__hawkowl hawkowl
cc__exarkun cc__hynek
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_1"></a>|@hynek commented|
|-|-|

+1 on deprecation.  They sound like they do something useful which they actually don’t and it keeps confusing the heck out of people. 

I have removed the milestone since they are only for regressions.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

I think "we don't recommend their use" is good enough justification. Tools like fpm now exist which makes it not required.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [44378]) Branching to deprecate-tap2whatever-7682.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Let's deprecate this so we can remove it sooner rather than later.

Pushed up, builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes looks good to merge. Thanks!


Just minor comment. I don't like this indentation as it requires tabs and then up to 3 extra spaces and backspaces...
```
+warnings.warn("tap2deb is deprecated since Twisted 15.2.",
+              category=DeprecationWarning, stacklevel=2)
+
```


I prefer 
```
warnings.warn(
    "tap2deb is deprecated since Twisted 15.2.",
    category=DeprecationWarning, stacklevel=2)
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44447]) Merge deprecate-tap2whatever-7682: Deprecate tap2deb and tap2rpm

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7682](https://github.com/twisted/twisted/issues/7682)
